metrics/sinks/gcm/gcm.go

```diff
@@ -321,21 +324,37 @@ func CreateGCMSink(uri *url.URL) (core.DataSink, error) {
 		return nil, fmt.Errorf("invalid metrics parameter: %s", metrics)
 	}
 
-	if err := gce_util.EnsureOnGCE(); err != nil {
-		return nil, err
+	var gcpConfig struct {
+		ProjectId string `json:"project_id"`
 	}
 
-	// Detect project ID
-	projectId, err := gce.ProjectID()
+	client, err := google.DefaultClient(oauth2.NoContext, gcm.MonitoringScope)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating oauth2 client: %s", err)
 	}
 
-	// Create Google Cloud Monitoring service.
-	client := oauth2.NewClient(oauth2.NoContext, google.ComputeTokenSource(""))
 	gcmService, err := gcm.New(client)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating GCM service: %s", err)
+	}
+
+	// Try and get the project ID, GCE metadata first
+	projectId, err := gce.ProjectID()
```

Comment:
Two comments here:
(a) You want the project id from the credentials to override the one from the environment,
(b) You probably still want a way to supply the project id explicitly, in case the credentials aren't owned by the target project but simply have access to it.